src/hooks/useActiveOnboard.ts

```diff
@@ -142,75 +129,73 @@ export function useActiveOnboard<T = any>():ActiveOnboardInterface<T> {
+        const { chainId } = IsSupportedChain(newChainId)
+        setChain({ chainId: chainId })
+        StoreOnboardState(currentWallet, chainId)
+        setActivated(true)
     }
-  }, [activated, tried, connect, previouslyConnected])
 
-  useEffect(() => {
-    const isConnected = connectedWallets.length > 0
-
-    if (isConnected && connectedChain){
-      updateStorage(connectedChain.id, connectedWallets)
+    const connectOnboard = async () => {
+        // Coinbase reloads instead of sending accountsChanged event, so empty storage if no active address can be found
+        if (previouslyConnected[0].label[0] === 'Coinbase Wallet') {
+            const isStillActive = localStorage.getItem('-walletlink:https://www.walletlink.org:Addresses')
+            if (!isStillActive) {
+                localStorage.removeItem('currentConnectWallet')
+                setTried(true)
+                return
+            }
+        }
+        // disableModals:true for silently connecting
+        await connect({ autoSelect: { label: previouslyConnected[0].label[0], disableModals: true } })
+        setActivated(true)
     }
 
-    // disconnect
-    if (!isConnected && previouslyConnected.length && (tried || activated)){
-      const toReload = WalletLabels.includes(previouslyConnected[0].label[0])
-      StoreOnboardState(connectedWallets, '0x1')
-      setActivated(false)
+    // eager connect
+    useEffect(() => {
+        if (previouslyConnected.length && !tried) {
+            connectOnboard()
+            setTried(true)
+        } else if (activated || !previouslyConnected[0]) {
+            setTried(true)
+        }
+    }, [activated, tried, connect, previouslyConnected])
+
+    useEffect(() => {
+        const isConnected = connectedWallets.length > 0
 
-      if (previouslyConnected[0].label[0] === 'Coinbase Wallet' && activated){
-        for (let i=0; i < WalletLinkKeys.length;i++) {
-          localStorage.removeItem(WalletLinkKeys[i])
-          window.location.reload()
+        if (isConnected && connectedChain) {
+            updateStorage(connectedChain.id, connectedWallets)
         }
-      }
 
-      if (toReload && activated) {
-        localStorage.removeItem('walletconnect')
-        window.location.reload() // temporarily necessary, as there is a irrecoverable error/bug when not reloading
-      }
-    }
-  }, [connectedWallets, tried]) 
+        // disconnect
+        if (!isConnected && previouslyConnected.length && (tried || activated)) {
+            const toReload = WalletLabels.includes(previouslyConnected[0].label[0])
+            StoreOnboardState(connectedWallets, '0x1')
+            setActivated(false)
+
+            if (previouslyConnected[0].label[0] === 'Coinbase Wallet' && activated) {
+                for (let i = 0; i < WalletLinkKeys.length; i++) {
+                    localStorage.removeItem(WalletLinkKeys[i])
+                    window.location.reload()
```

Comment:
pls reuse this function - it may be helpful and also allows you to set url from which app should be restarted

https://github.com/GoodDollar/GoodDAPP/blob/master/src/lib/utils/system.web.js#L3